### PR TITLE
Support transitionEnterTimeout & transitionLeaveTimeout

### PR DESCRIPTION
**Pull request by myarcane**
### Description
`transitionEnterTimeout` & `transitionLeaveTimeout` are new `<CSSTransitionGroup/>` props.
These props were already implemented in [the original react repository](https://github.com/reactjs/react-transition-group). So this is simply a minimal port of the react implementation. 

### Why ?
In many cases, the `transitionend` is not reliable (check https://github.com/developit/preact-css-transition-group/issues/8#issuecomment-288089541). Adding timeout is a way to solve the issue.    

### How to use ? 
```
<CSSTransitionGroup
   transitionName="example"
   transitionEnterTimeout={300}
   transitionLeaveTimeout={300}
   ...
 />

### CODE REVIEW

File: src/CSSTransitionGroupChild.js [hunk 1]
```diff
@@ -19,10 +19,12 @@ import { addEndEventListener, removeEndEventListener } from './TransitionEvents'
...
 
 export class CSSTransitionGroupChild extends Component {
-	transition(animationType, finishCallback) {
-		let node = getComponentBase(this),
-			className = this.props.name[animationType] || this.props.name + '-' + animationType,
-			activeClassName = this.props.name[animationType + 'Active'] || className + '-active';
+	transition(animationType, finishCallback, timeout) {
+		let node = getComponentBase(this);
+
+		let className = this.props.name[animationType] || this.props.name + '-' + animationType;
```
Comment (hassanbazzi) [inline src/CSSTransitionGroupChild.js:25]: is this neccesary?

Comment (developit) [reply src/CSSTransitionGroupChild.js:25]: Not sure what it's doing myself, but I am not super familiar with how this lib is used. Mybe @myarcane can give us some context :)

Comment (myarcane) [reply src/CSSTransitionGroupChild.js:25]: Hi @hassanbazzi & @developit : 
I didn't do anything particular here. I've just removed the grouped coma declaration. Nothing else !
And I think `className` is still needed 👉  https://github.com/myarcane/preact-css-transition-group/blob/663cd4e90dc8476290f955b5829d60ced9d89ff3/src/CSSTransitionGroupChild.js#L56

**Comment by myarcane** [comment]
@hassanbazzi & @developit : can we merge the PR ?

**Comment by trsh** [comment]
@developit please review !

**Review by developit — APPROVED** [review]
LGTM

**Comment by myarcane** [comment]
Thx @developit 👍

**Pull request merged**